clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -1745,51 +1742,78 @@ public void testSeekWithInFlightReset() {
         assertEquals(237L, subscriptions.position(tp0).offset);
     }
 
-    @Timeout(10)
+    private boolean listOffsetMatchesExpectedReset(
+        TopicPartition tp,
+        OffsetResetStrategy strategy,
+        AbstractRequest request
+    ) {
+        if (!(request instanceof ListOffsetsRequest)) {
+            return false;
```

Comment:
Why not throwing exception straightforwardly when receiving those unexpected requests?

Reply:
I pushed an update to turn these into assertions. I also filed https://issues.apache.org/jira/browse/KAFKA-12263.